evap/staff/forms.py

```diff
@@ -204,6 +213,14 @@ def validate_unique(self):
         except forms.ValidationError as e:
             self._update_errors(e)
 
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        responsibility = self.cleaned_data['responsibility']
+        self.instance.responsible = responsibility == Contribution.IS_RESPONSIBLE
+        self.instance.can_edit = (responsibility == Contribution.IS_RESPONSIBLE) or (responsibility == Contribution.IS_EDITOR)
+        self.instance.save()
+        return self.instance
```

Comment:
please do the stuff first and then do `return super().save(*args, **kwargs)` without `self.instance.save()`, i have the feeling that right now you would change values in the db even if the formset generates some errors.